File: CSharp/MasterDetail/BooleanToVisibilityConverter.cs
/*
 * Xceed DataGrid for WPF - SAMPLE CODE - Master/Detail Sample Application
 * Copyright (c) 2007-2024 Xceed Software Inc.
 * 
 * [BooleanToVisibilityConverter.cs]
 *  
 * This file demonstrates how to create a converter from a boolean
 * value to a Visibility value.
 * 
 * This file is part of the Xceed DataGrid for WPF product. The use 
 * and distribution of this Sample Code is subject to the terms 
 * and conditions referring to "Sample Code" that are specified in 
 * the XCEED SOFTWARE LICENSE AGREEMENT accompanying this product.
 *
 */

using System;
using System.Windows;
using System.Windows.Data;

namespace Xceed.Wpf.DataGrid.Samples.MasterDetail
{
  public class BooleanToVisibilityConverter : IValueConverter
  {
    public object Convert( object value, Type targetType, object parameter, System.Globalization.CultureInfo culture )
    {
      if( !targetType.IsAssignableFrom( typeof( Visibility ) ) )
        return DependencyProperty.UnsetValue;

      if( value == null || ( bool )value == false )
        return Visibility.Collapsed;

      return Visibility.Visible;
    }

    public object ConvertBack( object value, Type targetType, object parameter, System.Globalization.CultureInfo culture )
    {
      return Binding.DoNothing;
    }
  }
}

File: Visual Basic .NET/Tableflow/BooleanToVisibilityConverter.vb
'
'* Xceed DataGrid for WPF - SAMPLE CODE - Tableflow™ Sample Application
'* Copyright (c) 2007-2024 Xceed Software Inc.
 '*
 '* [BooleanToVisibilityConverter.cs]
 '*
 '* This file demonstrates how to create a converter from a boolean
 '* value to a Visibility value.
 '*
 '* This file is part of the Xceed DataGrid for WPF product. The use
 '* and distribution of this Sample Code is subject to the terms
 '* and conditions referring to "Sample Code" that are specified in
 '* the XCEED SOFTWARE LICENSE AGREEMENT accompanying this product.
 '*
 '


Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Text
Imports System.Windows.Data
Imports System.Windows

Namespace Xceed.Wpf.DataGrid.Samples.Tableflow
  Public Class BooleanToVisibilityConverter
	  Implements IValueConverter
	Public Function Convert(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As System.Globalization.CultureInfo) As Object Implements IValueConverter.Convert
	  If (Not targetType.IsAssignableFrom(GetType(Visibility))) Then
		Return DependencyProperty.UnsetValue
	  End If

	  If value Is Nothing OrElse CBool(value) = False Then
		Return Visibility.Collapsed
	  End If

	  Return Visibility.Visible
	End Function

	Public Function ConvertBack(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As System.Globalization.CultureInfo) As Object Implements IValueConverter.ConvertBack
	  Return Binding.DoNothing
	End Function
  End Class
End Namespace
